### PR TITLE
[Multiple Hardwares] [Rest-server] support different environmental variables for computing device when use hived 

### PR DESCRIPTION
**Pull request by hzy46**

### CODE REVIEW

File: src/rest-server/config/rest-server.yaml [hunk 1]
```diff
@@ -32,3 +32,11 @@ github-path: marketplace
...
 enable-priority-class: "true"
 sql-max-connection: 50
+# `hived-computing-device-envs` only works when hived scheduler is enabled.
+# It determines the environmental variables we inject into the container.
+# e.g. if hived-computing-device-envs is set to ['NVIDIA_VISIBLE_DEVICES', 'HIVED_VISIBLE_DEVICES'],
+# and device 0 and device 1 are used, the container will receive environmental variables:
+# NVIDIA_VISIBLE_DEVICES=0,1 and HIVED_VISIBLE_DEVICES=0,1
+hived-computing-device-envs:
+- 'NVIDIA_VISIBLE_DEVICES'
+- 'HIVED_VISIBLE_DEVICES'
```
Comment (abuccts) [inline src/rest-server/config/rest-server.yaml:42]: could you pass a comma separated string directly?

File: src/rest-server/src/config/launcher.js [hunk 2]
```diff
@@ -38,6 +38,7 @@ const k8sLauncherConfigSchema = Joi.object()
     enabledJobHistory: Joi.boolean().required(),
     writeMergerUrl: Joi.string().required(),
     defaultComputingDeviceType: Joi.string().required(),
+    hivedComputingDeviceEnvs: Joi.array().items(Joi.string()),
```
Comment (abuccts) [inline src/rest-server/src/config/launcher.js:41]: required?

File: src/rest-server/src/models/v2/job/k8s.js [hunk 3]
```diff
@@ -636,10 +636,12 @@ const generateTaskRole = (
   };
   // check cpu job
   if (config.taskRoles[taskRole].resourcePerInstance.gpu === 0) {
-    frameworkTaskRole.task.pod.spec.containers[0].env.push({
-      name: 'NVIDIA_VISIBLE_DEVICES',
-      value: 'none',
-    });
```
Comment (abuccts) [inline src/rest-server/src/models/v2/job/k8s.js:642]: this block is also used in default scheduler

https://github.com/microsoft/pai/blob/be2629950d9e298d9a5929a10118f4bd850095d6/src/rest-server/config/rest-server.yaml#L35

Comment (abuccts) [reply src/rest-server/src/models/v2/job/k8s.js:642]: maybe set its default value to `NVIDIA_VISIBLE_DEVICES` in default scheduler

Comment (hzy46) [reply src/rest-server/src/models/v2/job/k8s.js:642]: Done. Remains the same in default scheduler.

File: src/rest-server/src/config/launcher.js (hunk 2 above)
Comment (hzy46) [reply src/rest-server/src/config/launcher.js]: Done.

File: src/rest-server/config/rest-server.yaml (hunk 1 above)
Comment (hzy46) [reply src/rest-server/config/rest-server.yaml]: Done.

File: src/rest-server/config/rest-server.yaml [hunk 4]
```diff
@@ -32,3 +32,9 @@ github-path: marketplace
 debugging-reservation-seconds: 604800
 enable-priority-class: "true"
 sql-max-connection: 50
+# `hived-computing-device-envs` only works when hived scheduler is enabled.
+# It determines the environmental variables we inject into the container.
+# e.g. if hived-computing-device-envs is set to ['NVIDIA_VISIBLE_DEVICES', 'HIVED_VISIBLE_DEVICES'],
```
Comment (abuccts) [inline src/rest-server/config/rest-server.yaml:37]: update accordingly

File: src/rest-server/config/rest-server.yaml [hunk 5]
```diff
@@ -32,3 +32,9 @@ github-path: marketplace
...
+# and device 0 and device 1 are used, the container will receive environmental variables:
```
Comment (abuccts) [inline src/rest-server/config/rest-server.yaml:38]: environmental -> environment?

File: src/rest-server/config/rest-server.yaml [hunk 6]
```diff
@@ -32,3 +32,9 @@ github-path: marketplace
 debugging-reservation-seconds: 604800
 enable-priority-class: "true"
 sql-max-connection: 50
+# `hived-computing-device-envs` only works when hived scheduler is enabled.
+# It determines the environmental variables we inject into the container.
```
Comment (abuccts) [inline src/rest-server/config/rest-server.yaml:36]: environmental -> environment?
add a reference to https://github.com/microsoft/hivedscheduler/blob/master/doc/user-manual.md#scheduling-gpus

File: src/rest-server/src/config/launcher.js [hunk 7]
```diff
@@ -38,6 +38,7 @@ const k8sLauncherConfigSchema = Joi.object()
     enabledJobHistory: Joi.boolean().required(),
     writeMergerUrl: Joi.string().required(),
     defaultComputingDeviceType: Joi.string().required(),
+    hivedComputingDeviceEnvs: Joi.array().items(Joi.string()).required(),
```
Comment (abuccts) [inline src/rest-server/src/config/launcher.js:41]:
```suggestion
    hivedComputingDeviceEnvs: Joi.array().items(Joi.string()).min(1).required(),
```

**Review by abuccts — APPROVED** [review]

File: src/rest-server/config/rest-server.yaml (hunk 5 above)
Comment (hzy46) [reply src/rest-server/config/rest-server.yaml]: done

File: src/rest-server/config/rest-server.yaml (hunk 6 above)
Comment (hzy46) [reply src/rest-server/config/rest-server.yaml]: done

File: src/rest-server/src/config/launcher.js (hunk 7 above)
Comment (hzy46) [reply src/rest-server/src/config/launcher.js]: done

File: src/rest-server/config/rest-server.yaml (hunk 4 above)
Comment (hzy46) [reply src/rest-server/config/rest-server.yaml:37]: done

**Pull request merged**